shell-completion/Makefile

```diff
@@ -1,30 +1,34 @@
-
-.PHONY: command-options clean
+# Setting the number of job runners like this in the makefile only works in
+# GNU Make 4.3 or later. Older versions will require that either an env
+# variable be set before running or command line flag be passed at runtime to
+# get parallel jobs.
+MAKEFLAGS += --jobs=$(shell nproc 2>/dev/null || printf 8)
 
 all: hledger-completion.bash
 
-hledger-completion.bash: hledger-completion.bash.m4 commands-list.txt query-filters.txt generic-options.txt command-options
+COMMANDS := $(sort $(shell hledger | ./output-commands.sh | grep -v ^hledger | sort -u) ui web api)
+CMDOPTFILES := $(foreach CMD,$(COMMANDS),options-$(CMD).txt)
+
+hledger-completion.bash: hledger-completion.bash.m4 commands.txt commands-list.txt query-filters.txt generic-options.txt $(CMDOPTFILES)
 	m4 hledger-completion.bash.m4 > $@
 
 generic-options.txt:
 	hledger -h | ./output-options.sh | sort -u > $@
 
 commands.txt:
-	hledger | ./output-commands.sh | grep -v ^hledger | sort -u > $@
-	echo ui  >> $@
-	echo web >> $@
-	echo api >> $@
+	printf "%s\n" $(COMMANDS) > $@
 
-commands-list.txt: commands.txt
-	paste -sd, $^ | tr -d '\n' > $@
+commands-list.txt:
+	sed -ze 's/ /,/g;s/\n//' <<< "$(COMMANDS)" > $@
```

Comment:
I really don't want this to come out as nagging, but `-z` is a GNU extension and thus not portable. I believe portability should not be sacrificed if there are trivial ways to avoid that, so pick your poison and use something else please. And sorry for being such a pedantic, can't help myself... It works beautifully otherwise!

Reply:
And another suggestion to stay with our friends:
```make
commands-list.txt:
	printf "%s," $(COMMANDS) | sed 's/,$$//' > $@
```

Reply:
No no BSD compatibility is a must, that's not nagging. I just don't have one handy for testing.